mono/profiler/mono-profiler-log.c

```diff
@@ -4285,49 +4285,72 @@ new_filename (const char* filename)
 	return res;
 }
 
-static void*
-helper_thread (void* arg)
+static void
+add_to_fd_set (fd_set *set, int fd, int *max_fd)
+{
+	/*
+	 * This should only trigger for the basic FDs (server socket, pipe, perf
+	 * events) at startup if for some mysterious reason they're too large. In
+	 * this case, the profiler really can't function, and we're better off
+	 * printing an error and exiting.
+	 */
+	if (fd >= FD_SETSIZE) {
+		fprintf (stderr, "File descriptor is out of bounds for fd_set: %d\n", fd);
+		exit (1);
```

Comment:
We do. For example, if we can't set up the server socket for whatever reason, we `exit (1)`. We also do it for command line syntax errors, e.g. `log:sample=,alloc`. Whether that's good or bad is arguable, but I'm just following precedent here.

I don't think there's much value in continuing to execute if the profiler is in an unusable state, though. Note that this particular check will only fail at early startup if either the pipe or server socket FD is larger than `FD_SETSIZE` for some reason.